testdata/src/go.uber.org/maps/maps.go

```diff
@@ -692,3 +698,56 @@ func testExplicitBool(mp map[int]*int, i int) *int {
 	}
 	return &i
 }
+
+// tests for checking index expressions as constants both built-in and user defined (declared locally and in another package)
+// nonnil(mp, mp[])
+func testIndexAsUserDefinedConst(mp map[string]*string, i int) string {
```

Comment:
👍 (why don't we create a new package, maybe `consts`, instead of having it reside in `maps`?)

Reply:
Fair point. Done.